testsuite/classlibrary/TestNodeProxy.sc

```diff
@@ -3,139 +3,158 @@ TestNodeProxy : UnitTest {
 			"after rebuild and server running: resources should be on the server");
 
 		// QUIT
-		s.quit;
+		server.quit;
 		x = nil;
-		Ndef(\x).rebuild;
+		proxy.rebuild;
 
 		this.assert(x == "rebuilt",
 			"after server quit and rebuild: synth function should have been called");
 
-		this.assert(Ndef(\x).loaded == false,
+		this.assert(proxy.loaded == false,
 			"after server quit: no resources should be on the server");
 
 		/* will be fixed in master
-		Ndef(\x).send;
-		this.assert(Ndef(\x).loaded != true,
+		proxy.send;
+		this.assert(proxy.loaded != true,
 			"after server quit: send should not assume server resources loaded");
 		*/
 
 		// CLEAR
-		Ndef.clear;
-		try { Ndef(\x).ar(s.options.numAudioBusChannels * 2) };
+		proxy.clear;
+		try { proxy.ar(server.options.numAudioBusChannels * 2) };
 
-		this.assert(Ndef(\x).isNeutral,
+		this.assert(proxy.isNeutral,
 			"trying to allocate more bus channels than available should leave the node proxy untouched"
 		);
 
 		// ELASTIC
 
-		Ndef(\x).reshaping = \elastic;
+		proxy.reshaping = \elastic;
 
-		try { Ndef(\x, { DC.ar(0 ! (s.options.numAudioBusChannels * 2)) }) };
+		try { Ndef(\x, { DC.ar(0 ! (server.options.numAudioBusChannels * 2)) }) };
 
-		this.assert(Ndef(\x).isNeutral,
+		this.assert(proxy.isNeutral,
 			"when elastic: trying to allocate more bus channels than available should leave the node proxy untouched"
 		);
 
-		Ndef(\x).ar(8);
+		proxy.ar(8);
 
-		this.assert(Ndef(\x).numChannels == 8,
+		this.assert(proxy.numChannels == 8,
 			"setting number of channels should change them"
 		);
 
-		try { Ndef(\x, { DC.ar(0 ! (s.options.numAudioBusChannels * 2)) }) };
+		try { Ndef(\x, { DC.ar(0 ! (server.options.numAudioBusChannels * 2)) }) };
 
-		this.assert(Ndef(\x).numChannels == 8,
+		this.assert(proxy.numChannels == 8,
 			"when elasic and initialised: trying to allocate more bus channels than available should leave the node proxy untouched"
 		);
 
 
+		proxy.clear;
+		server.quit;
+		server.remove;
+
 	}
 
 	test_fadeTime {
 
-		var s = Server.default;
-		s.quit;
-		this.bootServer(s);
-		s.sync;
-		1.wait;
+		var server = Server(this.class.name);
+		var proxy;
+
+		server.bootSync;
 
-		Ndef.clear;
-		Ndef(\x).fadeTime = 2;
-		Ndef(\x).reshaping = \elastic;
-		Ndef(\x).clear(1);
+		proxy = NodeProxy(server);
+		proxy.fadeTime = 2;
+		proxy.reshaping = \elastic;
+		proxy.clear(1);
 		1.01.wait;
-		this.assert(Ndef(\x).isNeutral, "after fadeTime of clear, node proxy should be neutral again");
+
+		this.assert(proxy.isNeutral, "after fadeTime of clear, node proxy should be neutral again");
+
+		proxy.clear;
+		server.quit;
+		server.remove;
 
 	}
 
+	test_synthDefControl_build {
+
+		var server = Server(this.class.name);
+		var proxy;
+
+		server.bootSync;
+
+		proxy = NodeProxy(server);
+		proxy.source = { Silent.ar };
+
+		this.assert(proxy.objects.first.hasFadeTimeControl, "functions should register their fadeTime control");
+
+		proxy.clear;
```

Comment:
I thought I had left this comment somewhere else, but I can't find it -- add a `server.sync` call here, otherwise "FAILURE IN SERVER /g_new Group 1 not found" is printed.